tests/factory/test_factory_invalid_options.py

```diff
@@ -2,186 +2,151 @@
+        'platform': 'win32',
 
-            'url': 'http://localhost',
-            'version': '23.0.1',
-        },
-    },
+        'url': 'http://localhost',
+        'version': '23.0.1',
+    }),
 
     # TryScraper
-    {
-        'scraper_type': 'try',
-        'builds': ['/firefox/try-builds/test-user@mozilla.com-8fcac92cfcad/try-foobar/'],
-        'fname': '8fcac92cfcad-firefox-38.0a1.en-US.mac.dmg',
-        'kwargs': {
-            'revision': '8fcac92cfcad',
-            'platform': 'mac64',
-
-            'build_id': '20131001030204',
-            'build_number': '1',
-            'url': 'http://localhost',
-            'version': '23.0.1',
-        },
-    },
-]
-
-
-class TestFactoryMissingMandatoryOptions(mhttpd.MozHttpdBaseTest):
-
-    def test_release_without_version(self):
-        """Test that missing mandatory options for release builds raise an exception"""
-        self.assertRaises(ValueError, FactoryScraper,
-                          scraper_type='release',
-                          destination=self.temp_dir,
-                          base_url=self.wdir,
-                          logger=self.logger)
-
-    def test_candidate_without_version(self):
-        """Test that missing mandatory options for candidate builds raise an exception"""
-        self.assertRaises(ValueError, FactoryScraper,
-                          scraper_type='release',
-                          destination=self.temp_dir,
-                          base_url=self.wdir,
-                          logger=self.logger)
-
-    def test_try_without_revision(self):
-        """Test that missing mandatory options for try builds raise an exception"""
-        self.assertRaises(ValueError, FactoryScraper,
-                          scraper_type='try',
-                          destination=self.temp_dir,
-                          base_url=self.wdir,
-                          logger=self.logger)
-
-    def test_non_daily_fennec(self):
-        """Test that non-daily scrapper_type for fennec raises exception"""
-        self.assertRaises(NotSupportedError, FactoryScraper,
-                          scraper_type='candidate',
-                          destination=self.temp_dir,
-                          base_url=self.wdir,
-                          logger=self.logger,
-                          application='fennec',
-                          version='60.0b1')
-
-    def test_non_release_non_candidate_devedition(self):
-        """Test that non-relase and non-candidate scrapper type for devedition raises exception"""
-        self.assertRaises(NotSupportedError, FactoryScraper,
-                          scraper_type='daily',
-                          destination=self.temp_dir,
-                          base_url=self.wdir,
-                          logger=self.logger,
-                          application='devedition',
-                          version='60.0b1')
-
-
-class TestFactoryUnusedOptions(mhttpd.MozHttpdBaseTest):
```

Comment:
Please lets move all the tests from this class to it's own file, and name both files similar to the class names.

Reply:
separated them into a separate file you can take look here - https://github.com/rgpt/mozdownload/blob/Convert_directoryParser_factory_toPytest/tests/factory/test_factory_unused_options.py

Reply:
This is incorporated. Please verify and resolve